src/Http/WebUtilities/test/FileBufferingReadStreamTests.cs

```diff
@@ -486,5 +495,22 @@ private static string GetCurrentDirectory()
         {
             return AppContext.BaseDirectory;
         }
+
+        private class NumberOfWritesMemoryStream : MemoryStream
+        {
+            public int NumberOfWrites { get; set; }
+
+            public override void Write(byte[] buffer, int offset, int count)
```

Comment:
Maybe this should just derive from stream and write to a MemoryStream member. Then you won't have to worry about this breaking because base.Write() someday starts calling Write(ReadOnlySpan) or something like that.